src/components/common/YourCompany/StartDate.vue

```diff
@@ -0,0 +1,226 @@
+              </v-btn>
+            </template>
+            <v-btn text color="primary" id="more-changes-btn"
+              @click="onChangeClicked(); dropdown = false">
+              <v-icon small color="primary">mdi-pencil</v-icon>
+              <span>{{ editLabel }}</span>
+            </v-btn>
+          </v-menu>
+        </span>
+
+      </v-col>
+    </v-row>
+  </section>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop, Watch } from 'vue-property-decorator'
+import { Getter, Action } from 'vuex-class'
+import { CommonMixin, DateMixin } from '@/mixins/'
+import { DatePicker as DatePickerShared } from '@bcrs-shared-components/date-picker'
+import { ActionBindingIF } from '@/interfaces'
+
+@Component({
+  components: {
+    DatePickerShared
+  }
+})
+export default class StartDate extends Mixins(CommonMixin, DateMixin) {
+  // Global getters
+  @Getter isBenIaCorrectionFiling!: boolean
+  @Getter isFirmCorrectionFiling!: boolean
+  @Getter getCorrectionStartDate!: string
+  @Getter getBusinessFoundingDate!: string // actually date-time
+  @Getter getCurrentDate!: string
+  @Getter getCurrentJsDate!: Date
+
+  // Global setter
+  @Action setCorrectionStartDate!: ActionBindingIF
+  @Action setValidComponent!: ActionBindingIF
+
+  /** Whether to show invalid section styling. */
+  @Prop({ default: false })
+  readonly invalidSection: boolean
+
+  protected onEditMode = false
+  protected isCorrected = !!this.getCorrectionStartDate
+  protected newCorrectedStartDate = undefined // date is "Month Day, Year"
+
+  /** The minimum start date that can be entered (Up to 2 years ago today). */
+  protected get startDateMin (): Date {
+    const startDateMin = new Date(this.getCurrentJsDate)
+    startDateMin.setFullYear(startDateMin.getUTCFullYear() - 2)
+    startDateMin.setHours(0, 0, 0) // Set time to 0 for accurate Date Rules comparison
+
+    return startDateMin
+  }
+
+  /** The minimum start date string. */
+  protected get startDateMinStr (): string {
+    return this.dateToYyyyMmDd(this.startDateMin)
+  }
+
+  /** The maximum start date that can be entered (Up to 90 days from today). */
+  protected get startDateMax (): Date {
+    const startDateMax = new Date(this.getCurrentJsDate)
+    startDateMax.setDate(startDateMax.getUTCDate() + 90)
+    return startDateMax
+  }
+
+  /** The maximum start date string. */
+  protected get startDateMaxStr (): string {
+    return this.dateToYyyyMmDd(this.startDateMax)
+  }
+
+  /** The business date or business start date string. */
+  protected get businessStartDate (): string {
+    if (
+      this.isFirmCorrectionFiling ||
+      this.isFirmChangeFiling ||
+      this.isFirmConversionFiling
+    ) {
+      if (this.getCorrectionStartDate && this.isCorrected) {
+        return this.yyyyMmDdToPacificDate(this.getCorrectionStartDate)
+      } else if (this.getBusinessFoundingDate) {
+        return this.apiToPacificDateLong(this.getBusinessFoundingDate)
```

Comment:
The first thing I noticed is that the displayed date is wrong, eg:

![image](https://user-images.githubusercontent.com/10002889/181815122-dcee159b-62f3-4e3c-9bb9-59f99229ed5e.png)

This is because the Founding Date coming from the API is just a date in this case -- you can ignore the time portion and assume the date is already Pacific time zone. But confirm with a BE developer that this is only true for corrections or firms. (I think that, for alterations and/or BENs, this really is a date-time in UTC.)

It would be nice for Founding Date to have a single definition. Maybe the BE guys have some ideas about this and you guys can solve this. Failing that, use code comments to indicate what getBusinessFoundingDate is in this scenario (and also update what it means in recognitionDateTime() in YourCompany.vue).

Reply:
I think that the founding date from the API is ahead by one day https://github.com/bcgov/business-edit-ui/blob/85c9b72fed839d86a58298fb072a916e01be449e/src/mixins/date-mixin.ts#L192 from the documentation for the apiToPacificDateLong description.

Okay I will reach out to someone from the BE to find out if we have a way to have a single definition. Failing that I will add some comments clarifying what getBusinessFoundingDate, and recogntionDateTime is.

Reply:
> I think that the founding date from the API is ahead by one day

Yes, it is if you interpret it as a UTC date-time. But it's OK if you interpret it as "just a date". Confirm with BE dev that it is indeed just a date for firms and a UTC date-time for BENs (and coops too maybe?).

Reply:
Argus did confirm this. I think we will need a change to the date-mixin as there is no common method for handling this type of date. I think that is a shared component?

Reply:
I believe all UIs have their own Date Mixin. There is also a shared component one for those components when they are used locally (ie, Storybook).

We have `yyyyMmDdToDate`, `yyyyMmDdToPacificDate`, etc. I would just slice the string to drop the time portion (firms only).

Reply:
> Argus did confirm this. I think we will need a change to the date-mixin as there is no common method for handling this type of date. I think that is a shared component?

"2022-06-01T00:00:00+00:00" is actually the day before at 5pm Pacific -- not what is intended here. So I'm not concerned about converting this to a valid Date in a new mixin. Just hack off the time portion, then convert it as a date already in Pacific timezone. Also please leave code comments explaining what's going on here.

-> Straight dates (yyyy-mm-dd) should be interpreted as "a date in Pacific tz".

Reply:
Okay I will make these changes and add comments explaining the reasoning behind the changes.